src/RollbarServiceProvider.php

```diff
@@ -51,35 +51,50 @@ public function boot()
     public function register()
     {
         // Don't register rollbar if it is not configured.
-        if (! getenv('ROLLBAR_TOKEN') and ! $this->app['config']->get('services.rollbar')) {
+        if ($this->stop() === true) {
             return;
         }
 
         $app = $this->app;
 
         $this->app->singleton('Rollbar\RollbarLogger', function ($app) {
-            
+
             $defaults = [
                 'environment'  => $app->environment(),
                 'root'         => base_path(),
             ];
             $config = array_merge($defaults, $app['config']->get('services.rollbar', []));
             $config['access_token'] = getenv('ROLLBAR_TOKEN') ?: $app['config']->get('services.rollbar.access_token');
-            
+
             if (empty($config['access_token'])) {
                 throw new InvalidArgumentException('Rollbar access token not configured');
             }
-            
+
             \Rollbar\Rollbar::init($config);
-            
+
             return Rollbar::logger();
         });
 
         $this->app->singleton('Rollbar\Laravel\RollbarLogHandler', function ($app) {
-            
+
             $level = getenv('ROLLBAR_LEVEL') ?: $app['config']->get('services.rollbar.level', 'debug');
 
             return new RollbarLogHandler($app['Rollbar\RollbarLogger'], $app, $level);
         });
     }
+
+    /**
+     * Check if we should prevent the service from registering
+     *
+     * @return boolean
+     */
+    public function stop()
+    {
+        $level = getenv('ROLLBAR_LEVEL') ?: $this->app['config']->get('services.rollbar.level', null);
+        $hasLevel = empty($level) === false || strtolower($level) == 'none';
```

Comment:
Granted, to avoid specificity bugs and avoid triggering errors in different PHP versions, I would suggest a few additional changes to the `stop()` method:

```diff
    public function stop()
    {
        $level = getenv('ROLLBAR_LEVEL') ?: $this->app['config']->get('services.rollbar.level', null);
-        $hasLevel = empty($level) === false || strtolower($level) == 'none';
+        $hasLevel = empty($level) === false || strtolower($level) != 'none';
-        $hasToken = !empty(getenv('ROLLBAR_TOKEN'));
+        $hasToken = getenv('ROLLBAR_TOKEN');
        $hasConfig = $this->app['config']->has('services.rollbar');
-        return $hasLevel === false || $hasToken === false && $hasConfig === false;
+        return $hasLevel === false || ($hasToken === false && $hasConfig === false);
    }
```

Tests are passing for me after making these changes, for both PHP 5.6 and PHP 7.2

Reply:
Cool, I made the changes you suggested & tests are passing... but, its not actually doing what its supposed to do.
hang on...

Reply:
What specific scenario is failing in your use?